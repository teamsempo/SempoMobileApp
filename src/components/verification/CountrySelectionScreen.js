import React from 'react'
import { connect } from 'react-redux';
import {
    StyleSheet,
    View,
    Text,
    BackHandler
} from 'react-native'

import { Headline, Subheading, Caption, List, Divider,Snackbar, ActivityIndicator } from 'react-native-paper'


import Styles from '../../Styles';
import CountryPickerWrapper from "../CountryPickerWrapper";
import {updateKYCDetails, loadKYCApplicationRequest} from "../../reducers/kycApplicationReducer";
import Button from "../Button";
import GoBackHeader from "./GoBackHeader";
import { CenterLoadingSpinner } from "../common/CenterLoadingSpinner.js"
import {strings} from "../../../locales/i18n";

const mapStateToProps = (state) => {
    return {
        locale: state.locale,
        kycApplication: state.kycApplication,
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        updateKYCDetails: (payload) => dispatch(updateKYCDetails(payload)),
        loadKYCApplication: (query) => dispatch(loadKYCApplicationRequest({query}))
    };
};

class CountrySelectionScreen extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            cca2: 'AU',
            name: 'Press to choose',
            visible: false,
        }
    }

    componentDidMount() {
        this.props.loadKYCApplication({trulioo_countries: true});
    }

    _onCountryChange(value) {
        console.log('value',value)
        this.setState({ cca2: value.cca2, name: value.name })
    }

    _onBack() {
        this.props.updateKYCDetails({flowStep: this.props.kycApplication.kycDetails.flowStep -1})
    };

    _onNext() {
        if (this.state.name === 'Press to choose') {
            this.setState({visible: true});
            return
        }
        this.props.updateKYCDetails({flowStep: this.props.kycApplication.kycDetails.flowStep + 1, country: this.state.cca2})
    }

    render() {
        const { kycApplication } = this.props;

        var countries = kycApplication.kycApplicationState.trulioo_countries;
        if (typeof countries === "undefined") {
            countries = []
        }

        return (
            <View style={Styles.rootContainer}>

                <View style={{flex: 1}}>
                    <GoBackHeader goBack={() => this._onBack()}/>
                </View>

                {kycApplication.loadStatus.isRequesting ? <CenterLoadingSpinner /> : null}

                {kycApplication.loadStatus.isRequesting ? null : <View style={{flex: 10}}>

                    <View style={styles.TopContainer}>
                        <Headline>{strings('CountrySelectionScreen.Title')}</Headline>
                        <Subheading>{strings('CountrySelectionScreen.Subtitle')}</Subheading>
                    </View>

                    <Divider/>


                    <View style={styles.CountryWrapper}>
                        <CountryPickerWrapper
                            countryCodes={countries}
                            countryCode={this.state.cca2}
                            name={this.state.name}
                            withCallingCode={false}
                            withCountryNameButton={true}
                            withCallingCodeButton={false}
                            onSelect={value => this._onCountryChange(value)}
                        />
                    </View>

                    <View style={styles.BottomButton}>
                        <Button onPress={() => this._onNext()} buttonText={strings('Common.Next')}/>
                    </View>

                </View>}

                <Snackbar
                    visible={this.state.visible}
                    onDismiss={() => this.setState({ visible: false })}
                    action={{
                        label: strings('Common.OK'),
                        onPress: () => {
                            this.setState({visible: false})
                        },
                    }}
                >
                    {strings('CountrySelectionScreen.Prompt')}
                </Snackbar>

            </View>
        )
    }
}
export default connect(mapStateToProps, mapDispatchToProps)(CountrySelectionScreen);

const styles = StyleSheet.create({
    TopContainer: {
        paddingVertical: 20,
        paddingHorizontal: 20,
    },
    Icon: {
        margin: 8
    },
    CountryWrapper: {
        paddingHorizontal: 20,
        top: 120,
        position: 'absolute',
        zIndex: 99,
    },
    BottomButton: {
        flex: 1,
        position: 'absolute',
        width: '100%',
        padding: 20,
        bottom: 0
    }
});