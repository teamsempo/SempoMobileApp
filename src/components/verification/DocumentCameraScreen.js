'use strict';
import React, { Component } from 'react';
import {
    Dimensions,
    StyleSheet,
    View,
    TouchableNativeFeedback
} from 'react-native';
import { RNCamera } from 'react-native-camera';

import { Headline, Text } from 'react-native-paper'

import { connect } from "react-redux";
import { strings } from '../../../locales/i18n';
import {updateKYCDetails} from "../../reducers/kycApplicationReducer";
import Button from "../Button";
import GoBackHeader from "./GoBackHeader";
import { addSpaceToString } from '../../utils';

const mapStateToProps = (state) => {
    return {
        kycApplication: state.kycApplication,
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        updateKYCDetails: (payload) => dispatch(updateKYCDetails(payload)),
    };
};

class DocumentCameraScreen extends Component {
    constructor(props) {
        super(props);
        this.state = {
            singleImage: false,
            documentFront: true,
            isPictureTaken: false,
            uriDocumentFront: null,
            uriDocumentBack: null,
        }
    }

    componentDidMount() {
        if (this.props.kycApplication.kycDetails.documentType.toLowerCase() === 'passport') {
            // passport only needs one image.
            this.setState({singleImage: true})
        }
    }

    takePhoto = async () => {
        if (this.camera) {
            const options = {
                quality: 0.7,
                skipProcessing: true,
            };
            const data = await this.camera.takePictureAsync(options).then((data) => {
                this.setState({ isPictureTaken: true, [this.state.documentFront ? 'uriDocumentFront' : 'uriDocumentBack']: data.uri }, () => console.log(this.state));
            }).then(() => this.camera.pausePreview())
        }
    };

    _resetCamera() {
        this.setState({
            singleImage: false,
            documentFront: true,
            isPictureTaken: false,
            uriDocumentFront: null,
            uriDocumentBack: null,
        }, () => this.camera.resumePreview())
    }

    _goBack() {
        return this.state.isPictureTaken ? this._resetCamera() : this.props.updateKYCDetails({flowStep: this.props.kycApplication.kycDetails.flowStep - 1, uriDocumentBack: null, uriDocumentFront: null})
    }

    _onNext(e) {
        if (!this.state.singleImage && this.state.documentFront) {
            // singleImage is False and document is currently front
            this.setState({isPictureTaken: false, documentFront: false}, () => this.camera.resumePreview());
            return
        }

        this.props.updateKYCDetails({flowStep: this.props.kycApplication.kycDetails.flowStep + 1, documentBackUri: this.state.uriDocumentBack, documentFrontUri: this.state.uriDocumentFront})
    }

    maskFrame() {
        return {
            backgroundColor: this.state.isPictureTaken ? '#FFF' : 'rgba(1,1,1,0.6)',
        }
    };

    render() {
        const { isPictureTaken } = this.state;
        const { height, width } = Dimensions.get('window');
        const maskRowHeight = Math.round((height - 270) / 20);
        const maskColWidth = (width - 270) / 2;

        const documentType = addSpaceToString(this.props.kycApplication.kycDetails.documentType).toString();
        const documentSide = this.state.documentFront ? 'front' : 'back';

        return (
            <View style={styles.rootContainer}>
                <View style={styles.container}>
                    <GoBackHeader goBack={() => this._goBack()} dark={!isPictureTaken} {...this.props} />
                    <RNCamera
                        ref={ref => {
                            this.camera = ref;
                        }}
                        style={styles.preview}
                        type={RNCamera.Constants.Type.back}
                        flashMode={RNCamera.Constants.FlashMode.off}
                        barCodeTypes={[RNCamera.Constants.BarCodeType.qr]}
                        permissionDialogTitle={`${strings('CameraScreen.permissionDialog')}`}
                        permissionDialogMessage={`${strings('CameraScreen.permissionDialog')}`}
                    >
                        <View style={styles.maskOutter}>
                            <View style={[{ flex: maskRowHeight - 5, flexDirection: 'row', justifyContent: 'center', alignItems: 'center' }, styles.maskRow, this.maskFrame()]} >
                                <Headline style={{color: isPictureTaken ? '#000' : '#FFF', textAlign: 'center'}}>{strings('DocumentCameraScreen.Title', {type: documentType, side: documentSide})}</Headline>
                            </View>

                            <View style={[{ flex: 25 }, styles.maskCenter]}>
                                <View style={[{ width: maskColWidth }, this.maskFrame()]} />

                                <View style={styles.maskInner}/>

                                <View style={[{ width: maskColWidth }, this.maskFrame()]} />
                            </View>

                            <View style={[{ flex: maskRowHeight, justifyContent: 'center'}, styles.maskRow, this.maskFrame()]}>

                                {isPictureTaken ? <View style={styles.BottomButton}>
                                    <Button outline={true} onPress={() => this._resetCamera()} buttonText={strings('DocumentCameraScreen.Retake')}/>
                                    <Button onPress={() => this._onNext()} buttonText={strings('DocumentCameraScreen.Readable', {type: documentType})}/>
                                </View> : null}

                                {isPictureTaken ? null : <Text style={{color: '#FFF', textAlign: 'center', paddingHorizontal: 10}}>{strings('DocumentCameraScreen.Prompt', {side: documentSide, type: documentType})}</Text>}

                                {isPictureTaken ? null : <TouchableNativeFeedback background={TouchableNativeFeedback.SelectableBackground()} onPress={() => this.takePhoto()}>
                                    <View style={{position: 'absolute', bottom: height - (height * .97), alignItems: 'center', width: width, zIndex: 99,paddingHorizontal: 22}}>
                                        <View style={{
                                            width: 24,
                                            borderRadius: 100,
                                            borderColor: '#bfbfbf',
                                            borderWidth: 24
                                        }}/>
                                        <View style={{
                                            position: 'absolute',
                                            top: 4,
                                            width: 18,
                                            borderRadius: 100,
                                            borderColor: '#FFF',
                                            borderWidth: 20
                                        }}/>
                                    </View>
                                </TouchableNativeFeedback>}

                            </View>
                        </View>
                    </RNCamera>
                </View>
            </View>
        );
    }
}
export default connect(mapStateToProps, mapDispatchToProps)(DocumentCameraScreen);

const styles = StyleSheet.create({
    rootContainer: {
        flex: 1,
        backgroundColor: '#FFF',
    },
    preview: {
        flex: 11,
    },
    container: {
        flex: 1,
    },
    cameraView: {
        flex: 1,
        justifyContent: 'flex-start',
    },
    maskOutter: {
        position: 'absolute',
        top: 0,
        left: 0,
        width: '100%',
        height: '100%',
        alignItems: 'center',
        justifyContent: 'space-around',
    },
    maskInner: {
        width: Dimensions.get('window').width * .9,
        backgroundColor: 'transparent',
        borderColor: 'white',
        borderWidth: 1,
    },
    maskFrame: {
        backgroundColor: 'rgba(1,1,1,0.6)',
    },
    maskRow: {
        width: '100%',
    },
    maskCenter: {
        flexDirection: 'row',
    },
    modalBackground: {
        flex: 1,
        alignItems: 'center',
        flexDirection: 'column',
        justifyContent: 'space-around',
        backgroundColor: '#00000040',
    },
    BottomButton: {
        flex: 1,
        position: 'absolute',
        width: '100%',
        padding: 20,
        bottom: 0
    }
});