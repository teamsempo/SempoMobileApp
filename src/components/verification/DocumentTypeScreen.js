import React from 'react'
import { connect } from "react-redux";
import {
    StyleSheet,
    View
} from 'react-native'

import { Headline, Subheading, List, Divider } from 'react-native-paper'

import Icon from 'react-native-vector-icons/MaterialCommunityIcons';

import Styles from '../../Styles';
import {loadKYCApplicationRequest, updateKYCDetails } from "../../reducers/kycApplicationReducer";
import GoBackHeader from "./GoBackHeader";
import { CenterLoadingSpinner } from "../common/CenterLoadingSpinner.js"
import { addSpaceToString } from "../../utils";
import {strings} from "../../../locales/i18n";

const mapStateToProps = (state) => {
    return {
        kycApplication: state.kycApplication,
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        updateKYCDetails: (payload) => dispatch(updateKYCDetails(payload)),
        loadKYCApplication: (query) => dispatch(loadKYCApplicationRequest({query}))
    };
};

class DocumentTypeScreen extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
        }
    }

    componentDidMount() {
        this.props.loadKYCApplication({trulioo_documents: true, country: this.props.kycApplication.kycDetails.country})
    }

    getIconName(document) {
        var possibleDocuments = ["Passport", "DrivingLicence"];
        var possibleIcons = {"Passport": "passport", "DrivingLicence":"car"};

        if (possibleDocuments.indexOf(document) !== -1) {
            return possibleIcons[document]
        } else {
            return 'account-card-details'
        }
    }

    _onBack() {
        this.props.updateKYCDetails({flowStep: this.props.kycApplication.kycDetails.flowStep - 1, documentType: null})
    }

    onNext(type) {
        this.props.updateKYCDetails({flowStep: this.props.kycApplication.kycDetails.flowStep + 1, documentType: type})
    }

    render() {
        const { kycApplication } = this.props;

        let documents = kycApplication.kycApplicationState.trulioo_documents;

        var documentTypes = [];
        if (typeof documents !== "undefined") {
            if (typeof documents[kycApplication.kycDetails.country] !== 'undefined'){
                documentTypes = documents[kycApplication.kycDetails.country];
            }
        }

        return (
            <View style={Styles.rootContainer}>

                <View style={{flex: 1}}>
                    <GoBackHeader goBack={() => this._onBack()}/>
                </View>

                {kycApplication.loadStatus.isRequesting ? <CenterLoadingSpinner /> : null}

                {kycApplication.loadStatus.isRequesting ? null : <View style={{flex: 10}}>
                    <View style={styles.TopContainer}>
                        <Headline>{strings('DocumentTypeScreen.Title')}</Headline>
                        <Subheading>{strings('DocumentTypeScreen.Subtitle')}</Subheading>
                    </View>

                    <Divider/>

                    <List.Section>
                        {documentTypes.map((item, index) => {
                            return (
                                <List.Item
                                    key={index}
                                    title={addSpaceToString(item)}
                                    left={() => <Icon style={styles.Icon} name={this.getIconName(item)} size={30} color="#000"/>}
                                    onPress={() => this.onNext(item)}
                                />
                            )
                        })}
                    </List.Section>
                </View>}

            </View>
        )
    }
}
export default connect(mapStateToProps, mapDispatchToProps)(DocumentTypeScreen);


const styles = StyleSheet.create({
    TopContainer: {
        paddingVertical: 20,
        paddingHorizontal: 20,
    },
    Icon: {
        margin: 8
    },
    CountryWrapper: {
        paddingHorizontal: 20,
        top: 120,
        position: 'absolute',
        zIndex: 99,
    }
});