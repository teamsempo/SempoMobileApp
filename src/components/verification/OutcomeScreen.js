import React from 'react'
import { connect } from 'react-redux';
import {
    StyleSheet,
    View,
    ActivityIndicator,
    Text,
    Image,
    Dimensions,
    ScrollView,
    Linking,
    BackHandler
} from 'react-native'

import { Headline, Subheading, Divider } from 'react-native-paper'


import Styles from '../../Styles';
import {createKYCApplication, updateKYCDetails} from "../../reducers/kycApplicationReducer";
import Button from "../Button";
import {CenterLoadingSpinner} from "../common/CenterLoadingSpinner";
import {strings} from "../../../locales/i18n";

const mapStateToProps = (state) => {
    return {
        locale: state.locale,
        kycApplication: state.kycApplication,
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        updateKYCDetails: (payload) => dispatch(updateKYCDetails(payload)),
        createKYCApplication: (body) => dispatch(createKYCApplication({body})),
    };
};

class OutcomeScreen extends React.Component {
    constructor(props) {
        super(props);
        this.state = {

        }
    }

    componentWillMount() {
        BackHandler.addEventListener('hardwareBackPress', this._onNext);
    }

    componentWillUnmount() {
        BackHandler.removeEventListener('hardwareBackPress', this._onNext);
    }

    _onNext() {
        // reset local state and flow.
        this.props.updateKYCDetails({
            flowStep: 1,
            country: null,
            documentType: null,
            documentFrontBase64: null, // image
            documentBackBase64: null,
            selfieBase64: null, // image
        });
        this.props.navigation.goBack();
    }

    render() {
        let { kycApplication } = this.props;
        let businessFormLink = 'https://drive.google.com/file/d/1SF-mZEa4nxH3yxR0ecuuNLtdzH11f5Al/view?usp=sharing';
        let isBusiness = kycApplication.kycApplicationState.type === 'BUSINESS';
        let isBusinessPrompt = isBusiness ? strings('OutcomeScreen.Business') : strings('OutcomeScreen.Subtitle');
        const { height, width } = Dimensions.get('window');

        let isRequesting = kycApplication.createStatus.isRequesting || kycApplication.editStatus.isRequesting;
        let success = kycApplication.createStatus.success || kycApplication.editStatus.success;
        var error = null;

        if (!isRequesting && !success) {
            // some error has occurred
            if (kycApplication.createStatus.error !== null) {
                error = kycApplication.createStatus.error.message
            } else {
                error = kycApplication.editStatus.error.message
            }
        }

        return (
            <View style={Styles.rootContainer}>

                {isRequesting ? <CenterLoadingSpinner loadingText={strings('SendPaymentCameraScreen.Sending')}/>: null}

                {isRequesting ? null : <View style={{flex: 10}}>

                    <View style={{alignItems: 'center'}}>

                        <Image
                            source={require('../img/identity_documents.png')}
                            style={{width: width / 1.5, height: height / 3}}
                            resizeMode='contain'
                        />

                    </View>

                    <View style={styles.TopContainer}>
                        <Headline>{strings('OutcomeScreen.Title', {outcome: success ? strings('OutcomeScreen.Pending') : strings('OutcomeScreen.Error')})}</Headline>
                        <Subheading>{success
                            ? isBusinessPrompt
                            : error}</Subheading>
                        {isBusiness ? <View style={{padding: 5}}>
                            <Button outline={true} onPress={() => Linking.openURL(businessFormLink)} buttonText={strings('OutcomeScreen.Form')}/>
                        </View> : null}
                    </View>

                    <Divider />

                    <View style={styles.BottomButton}>
                        <Button onPress={() => this._onNext()} buttonText={strings('Common.OK')}/>
                    </View>

                </View>}

            </View>
        )
    }
}
export default connect(mapStateToProps, mapDispatchToProps)(OutcomeScreen);

const styles = StyleSheet.create({
    TopContainer: {
        paddingVertical: 20,
        paddingHorizontal: 20,
    },
    BottomButton: {
        flex: 1,
        position: 'absolute',
        width: '100%',
        padding: 20,
        bottom: 0
    }
});