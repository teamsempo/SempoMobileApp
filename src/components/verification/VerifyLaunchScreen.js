import React from 'react'
import { connect } from 'react-redux'
import {
    StyleSheet,
    View,
    Image,
    Dimensions,
    ScrollView,
    Linking
} from 'react-native'

import { Headline, Subheading, Caption, List, Divider, Banner, Checkbox } from 'react-native-paper'
import Intercom from 'react-native-intercom';

import Icon from 'react-native-vector-icons/MaterialCommunityIcons';

import Styles from '../../Styles';
import Button from "../Button";
import { updateKYCDetails } from "../../reducers/kycApplicationReducer";
import GoBackHeader from "./GoBackHeader";
import {strings} from "../../../locales/i18n";

const mapStateToProps = (state) => {
    return {
        locale: state.locale,
        kycApplication: state.kycApplication,
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        updateKYCDetails: (payload) => dispatch(updateKYCDetails(payload)),
    };
};

class VerifyLaunchScreen extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            hasStartedFlow: false,
            errorBanner: false,
            promptBanner: false,
            isBusiness: false,
        }
    }

    componentWillMount() {
        let kyc_state = this.props.kycApplication.kycApplicationState;
        if (kyc_state !== null) {
            if (kyc_state.kyc_actions !== null && typeof kyc_state.kyc_actions !== "undefined") {
                this.setState({errorBanner: true});
                return
            }
            if (Object.keys(kyc_state).length > 0) {
                // else, set a prompt banner on second attempts
                this.setState({promptBanner: true})
            }
        }
    }

    _onNext() {
        let kyc_type = this.state.isBusiness ? 'BUSINESS' : 'INDIVIDUAL';
        this.props.updateKYCDetails({flowStep: this.props.kycApplication.kycDetails.flowStep + 1, account_type: kyc_type})
    }

    _getErrorMessage() {
        let kyc_actions = this.props.kycApplication.kycApplicationState.kyc_actions;
        if (kyc_actions !== null && typeof kyc_actions !== "undefined") {
            return strings('VerifyLaunchScreen.ErrorBanner') + '\n \n' + strings(`VerifyFailureOptions.${kyc_actions[0]}`)
        }
        return strings('VerifyLaunchScreen.ErrorBanner')
    }

    render() {
        const { height, width } = Dimensions.get('window');

        return (
            <View style={Styles.rootContainer}>

                <View style={{flex: 1}}>
                    <GoBackHeader {...this.props} />
                </View>

                <View style={{flex: 10}}>

                    <Banner
                        visible={this.state.errorBanner || this.state.promptBanner}
                        actions={[
                            {
                                label: this.state.errorBanner ? strings('Common.ContactSupport') : strings('VerifyLaunchScreen.LearnMore'),
                                onPress: () => this.state.errorBanner ? Intercom.displayMessenger() : Linking.openURL('https://help.withsempo.com/en/articles/3211219-what-is-a-kyc-check-and-who-needs-to-do-it'),
                            },
                            {
                                label: strings('Common.OK'),
                                onPress: () => this.setState({ errorBanner: false, promptBanner: false }),
                            },
                        ]}
                        image={({ size }) =>
                            <Icon style={styles.Icon} name={this.state.errorBanner ? 'alert-outline' : 'account-question'} size={size} color={this.state.errorBanner ? '#ff7a64' : '#42b1b1'}/>
                        }
                    >
                        {this.state.errorBanner ? this._getErrorMessage() : strings('VerifyLaunchScreen.PromptBanner')}
                    </Banner>

                    <ScrollView>

                        <View style={{alignItems: 'center'}}>

                            <Image
                                source={require('../img/identity_documents.png')}
                                style={{width: width / 1.5, height: height / 3}}
                                resizeMode='contain'
                            />

                        </View>

                        <View style={styles.TopContainer}>
                            <Headline>{strings('VerifyLaunchScreen.Title')}</Headline>
                            <Subheading>{strings('VerifyLaunchScreen.Subtitle')}</Subheading>
                            <Caption>{strings('VerifyLaunchScreen.LegalTerms')}<Caption style={{textDecorationLine: "underline"}} onPress={() => this.props.navigation.navigate('Terms', { title: strings('GoBackNav.DefaultTitleText')})}>{strings('VerifyLaunchScreen.Terms')}</Caption></Caption>
                        </View>

                        <Divider/>

                        <View style={styles.List}>
                            <List.Section>
                                <List.Subheader>{strings('VerifyLaunchScreen.HowHeading')}</List.Subheader>
                                <List.Item
                                    title={strings('VerifyLaunchScreen.StepOne')}
                                    left={props => <Icon style={styles.Icon} name={'numeric-1'} size={30} color="#000"/>}
                                />
                                <List.Item
                                    title={strings('VerifyLaunchScreen.StepTwo')}
                                    left={props => <Icon style={styles.Icon} name={'numeric-2'} size={30} color="#000"/>}
                                />
                                <Divider/>
                                <List.Item
                                    title={strings('VerifyLaunchScreen.BusinessQuestion')}
                                    left={props => <View style={{padding: 6}}><Checkbox
                                        status={this.state.isBusiness ? 'checked' : 'unchecked'}
                                        onPress={() => { this.setState({ isBusiness: !this.state.isBusiness }); }}
                                    /></View>}
                                />
                            </List.Section>
                        </View>

                    </ScrollView>

                    <View style={styles.BottomButton}>
                        <Button onPress={() => this._onNext()} buttonText={strings('Common.Start')}/>
                    </View>
                </View>

            </View>
        )
    }
}
export default connect(mapStateToProps, mapDispatchToProps)(VerifyLaunchScreen);

const styles = StyleSheet.create({
    CloseContainer: {
        position: 'absolute',
        left: 0,
        zIndex: 999
    },
    TopContainer: {
        paddingVertical: 20,
        paddingHorizontal: 20,
    },
    Icon: {
        margin: 8
    },
    List: {
        paddingBottom: 100
    },
    BottomButton: {
        // backgroundColor: '#FFF',
        flex: 1,
        position: 'absolute',
        width: '100%',
        padding: 20,
        bottom: 0
    }
});