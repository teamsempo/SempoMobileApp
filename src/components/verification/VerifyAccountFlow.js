import React from 'react'
import { connect } from 'react-redux';
import {
    StyleSheet,
    View,
    Dimensions,
} from 'react-native'

import VerifyLaunchScreen from "./VerifyLaunchScreen"
import CountrySelection from "./CountrySelectionScreen";
import DocumentType from "./DocumentTypeScreen";
import DocumentCameraScreen from "./DocumentCameraScreen";
import SelfieCameraScreen from "./SelfieCameraScreen";
import OutcomeScreen from "./OutcomeScreen";

const mapStateToProps = (state) => {
    return {
        locale: state.locale,
        kycApplication: state.kycApplication,
    };
};

class VerifyAccountFlow extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            total: 6, // number of steps in KYC flow
        }
    }

    static navigationOptions = ({ navigation }) => ({
        title: `${navigation.state.params.title}`,
    });

    render() {
        let { kycApplication } = this.props;

        let flowStep = kycApplication.kycDetails.flowStep;

        var content = null;

        if (flowStep === 1) {
            // launch screen, terms
            content = <VerifyLaunchScreen {...this.props} />;

        } else if (flowStep === 2) {
            // choose country
            content = <CountrySelection />;

        } else if (flowStep === 3) {
            // choose document type
            content = <DocumentType />;

        } else if (flowStep === 4) {
             // camera screen (front and back here)
            content = <DocumentCameraScreen/>;

        } else if (flowStep === 5) {
             // selfie screen
            content = <SelfieCameraScreen />
        } else if (flowStep === 6) {
            // review screen
            content = <OutcomeScreen {...this.props} />
        }

        return(
            <View style={{flex: 1}}>

                <View style={styles.slide}>
                    {content}
                </View>

            </View>
        );
    }
}
export default connect(mapStateToProps, null)(VerifyAccountFlow);

const styles = StyleSheet.create({
    slide: {
        flex: 1, // Take up all screen
        width: Dimensions.get('window').width,
    }
});