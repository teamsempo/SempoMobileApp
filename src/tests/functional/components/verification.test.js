import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store'

import VerifyLaunchScreen from '../../../components/verification/VerifyLaunchScreen';
import SelfieCameraScreen from '../../../components/verification/SelfieCameraScreen';
import VerifyAccountFlow from '../../../components/verification/VerifyAccountFlow';
import OutcomeScreen from '../../../components/verification/OutcomeScreen';
import DocumentTypeScreen from '../../../components/verification/DocumentTypeScreen';
import DocumentCameraScreen from '../../../components/verification/DocumentCameraScreen';
import CountrySelectionScreen from '../../../components/verification/CountrySelectionScreen';


describe('VerifyLaunchScreen', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const initialState = { }
            const mockStore = configureStore(initialState);
            const component = shallow(<VerifyLaunchScreen store={mockStore} />).dive();
            expect(component).toMatchSnapshot()
        });
    });
});

describe('SelfieCameraScreen', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const initialState = { }
            const mockStore = configureStore(initialState);
            const component = shallow(<SelfieCameraScreen store={mockStore} />).dive();
            expect(component).toMatchSnapshot()
        });
    });
});

describe('VerifyAccountFlow', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const initialState = { }
            const mockStore = configureStore(initialState);
            const component = shallow(<VerifyAccountFlow store={mockStore} />).dive();
            expect(component).toMatchSnapshot()
        });
    });
});

describe('OutcomeScreen', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const initialState = { }
            const mockStore = configureStore(initialState);
            const component = shallow(<OutcomeScreen store={mockStore} />).dive();
            expect(component).toMatchSnapshot()
        });
    });
});

describe('DocumentTypeScreen', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const initialState = { }
            const mockStore = configureStore(initialState);
            const component = shallow(<DocumentTypeScreen store={mockStore} />).dive();
            expect(component).toMatchSnapshot()
        });
    });
});

describe('DocumentCameraScreen', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const initialState = { }
            const mockStore = configureStore(initialState);
            const component = shallow(<DocumentCameraScreen store={mockStore} />).dive();
            expect(component).toMatchSnapshot()
        });
    });
});

describe('CountrySelectionScreen', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const initialState = { }
            const mockStore = configureStore(initialState);
            const component = shallow(<CountrySelectionScreen store={mockStore} />).dive();
            expect(component).toMatchSnapshot()
        });
    });
});
