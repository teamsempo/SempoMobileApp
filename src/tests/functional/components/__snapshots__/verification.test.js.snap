// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountrySelectionScreen Rendering should match to snapshot 1`] = `
<CountrySelectionScreen
  loadKYCApplication={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  updateKYCDetails={[Function]}
/>
`;

exports[`DocumentCameraScreen Rendering should match to snapshot 1`] = `
<DocumentCameraScreen
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  updateKYCDetails={[Function]}
/>
`;

exports[`DocumentTypeScreen Rendering should match to snapshot 1`] = `
<DocumentTypeScreen
  loadKYCApplication={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  updateKYCDetails={[Function]}
/>
`;

exports[`OutcomeScreen Rendering should match to snapshot 1`] = `
<OutcomeScreen
  createKYCApplication={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  updateKYCDetails={[Function]}
/>
`;

exports[`SelfieCameraScreen Rendering should match to snapshot 1`] = `
<SelfieCameraScreen
  createKYCApplication={[Function]}
  editKYCApplicationRequest={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  updateKYCDetails={[Function]}
/>
`;

exports[`VerifyAccountFlow Rendering should match to snapshot 1`] = `
<VerifyAccountFlow
  dispatch={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;

exports[`VerifyLaunchScreen Rendering should match to snapshot 1`] = `
<VerifyLaunchScreen
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  updateKYCDetails={[Function]}
/>
`;
